### ISSUE TITLE
Better handling for broken snippets

### ISSUE THREAD

**Issue by ivanalejandro0** [SirVer/ultisnips#858]
I get here because of https://github.com/honza/vim-snippets/issues/933 and https://github.com/SirVer/ultisnips/issues/857.
I realized that this had happened a lot over time (see links at the bottom) and I think that these kind of errors should be handled differently than just leaving the user with an unusable editor.
Basically, a problem like this with a snippets provider or a syntax error somewhere in any of your snippets forces you to take some action right away instead of continue using the editor.

Similar problems: https://github.com/SirVer/ultisnips/issues/220, https://github.com/SirVer/ultisnips/issues/223, https://github.com/SirVer/ultisnips/issues/224, https://github.com/SirVer/ultisnips/issues/227, https://github.com/SirVer/ultisnips/issues/266, https://github.com/SirVer/ultisnips/issues/317, https://github.com/SirVer/ultisnips/issues/768
https://github.com/SirVer/ultisnips/issues/857, https://github.com/honza/vim-snippets/issues/440, https://github.com/honza/vim-snippets/issues/933

(This was just a quick search, sorry if I mentioned someone not relevant)

This is really the first time that I have a problem with ultisnips, it's been great so far, thank you!

**Comment by markstos** [comment]
Yes.  I thought "~/.vim/snippets" would a natural place to put my UltiSnips.  I couldn't make sense of the stacktrace I was getting, because my syntax appeared correct. It would be very helpful if the error message  provided a hint that the file was parsed in "snipmate" syntax and not "UltiSnips" syntax.  Something like:

"Attempt to parse a file in a directory named 'snippets' as SnipMate syntax failed. If you want use UltiSnip syntax, please use a directory name other than 'snippets'". 

It didn't occur to me that the file might be parsed in a different format-- I certainly got no warning about the format when using `:UltiSnipEdit` to edit a file in a directory named `./snippet`. Without realizing what the problem is, it's hard to find the correct help. I was only lucky that a search landed me on this issue.

**Comment by SirVer** [comment]
@markstos comment is referring to a different problem than OP.

OPs problem is fixed at c309637e0243b71f7e3265ebe3cffdfe9674cfbc, since the engine is no longer hyper aggressively reloading snippet files (only after they have been saved inside vim and the next snippet is expanded).

**Comment by markstos** [comment]
Should I open a separate issue?

**Comment by SirVer** [comment]
@markstos Yes, if you care about this. Even better would be a PR with a fix :)
